modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxManagerImpl.java

```diff
@@ -342,6 +344,8 @@ public CompletableFuture<Void> finish(
             // If there are no enlisted groups, just update local state - we already marked the tx as finished.
             updateTxMeta(txId, old -> new TxStateMeta(commit ? COMMITTED : ABORTED, localNodeId, commitPartition, commitTimestamp(commit)));
 
+            decrementRwTxCount(txId);
```

Comment:
Finish might be called multiple times for the same transaction, thus I'd say that it would be better to handle decrement in an idempotent manner.

Reply:
fix it